### PR TITLE
support for ssh password

### PR DESCRIPTION
**Pull request by tgbloks**
* the networkservice resource has a new password property - if it is set
  the sshdriver will login with sshpass (needs to be installed) instead
  of the key file

Signed-off-by: Tobi Gschwendtner <tg@bloks.de>

### CODE REVIEW

File: labgrid/driver/sshdriver.py [hunk 1]
```diff
@@ -104,7 +114,7 @@ def run(self, cmd):
         returns:
         (stdout, stderr, returncode)
         """
-        complete_cmd = "ssh -x {prefix} {user}@{host} {cmd}".format(
+        complete_cmd = "ssh -x{prefix} {user}@{host} {cmd}".format(
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: As above I'm wondering about the change here.

File: labgrid/driver/sshdriver.py [hunk 2]
```diff
@@ -30,8 +30,10 @@ def __attrs_post_init__(self):
         self.logger = logging.getLogger("{}({})".format(self, self.target))
 
     def on_activate(self):
-        self.ssh_prefix = "-i {}".format(os.path.abspath(self.keyfile)
+        self.ssh_prefix = " -i {}".format(os.path.abspath(self.keyfile)
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: Is this change with the change below necessary? Looks like an unnecessary change to me.

File: labgrid/driver/sshdriver.py (hunk 1 above)
Comment (tgbloks) [reply labgrid/driver/sshdriver.py]: i found these changes necessary to prevent 2 spaces following each other in the cmd line.
(2 spaces cause split(' ') to add an empty element in the list which will be treated by ssh as the hostname...)

**Comment by tgbloks** [comment]
was waiting with the documentation till we agreed on the rest ;-)

File: labgrid/driver/sshdriver.py (hunk 1 above)
Comment (Emantor) [reply labgrid/driver/sshdriver.py]: Ah, I see. In that case I would like you too shuffle the `self.ssh_prefix += " -o LogLevel=ERROR"` line to the top and handle the extra missing space there. Otherwise we would have to touch the get and put functions as well. I tested your changes against a local device, works fine :)

Comment (tgbloks) [reply labgrid/driver/sshdriver.py]: damn - i missed put and get.
should work now with one change less ;-)
